### PR TITLE
Add missing entry in the changelog about the KOKKOS_DEDUCTION_GUIDE macro

### PR DESCRIPTION
**Pull request by tpadioleau**

### CODE REVIEW

**Comment by tpadioleau** [comment]
Shall I also add an entry in this page of the wiki https://kokkos.org/kokkos-core-wiki/API/core/Macros.html?highlight=macros ?

I think it should belong next to `KOKKOS_FUNCTION`/`KOKKOS_INLINE_FUNCTION` and `KOKKOS_DEFAULTED_FUNCTION` but I don't think they are listed.

**Comment by masterleinad** [comment]
> Shall I also add an entry in this page of the wiki [kokkos.org/kokkos-core-wiki/API/core/Macros.html?highlight=macros](https://kokkos.org/kokkos-core-wiki/API/core/Macros.html?highlight=macros) ?

See https://github.com/kokkos/kokkos-core-wiki/pull/578.

**Review by aprokop — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -83,6 +83,7 @@
 
 ### Bug Fixes
 * Do not return a copy of the input functor in `Experimental::for_each` [\#6910](https://github.com/kokkos/kokkos/pull/6910)
+* Introduce KOKKOS_DEDUCTION_GUIDE macro to allow user-defined deduction guide in device code for clang compiler [\#6954](https://github.com/kokkos/kokkos/pull/6954)
```
Comment (masterleinad) [inline CHANGELOG.md:86]:
```suggestion
* Introduce KOKKOS_DEDUCTION_GUIDE macro to allow for portable deduction guides [\#6954](https://github.com/kokkos/kokkos/pull/6954)
```

Comment (tpadioleau) [reply CHANGELOG.md]: see 88f490276a8ebe94ec2d2a2f747543854157d22f

**Review by masterleinad — APPROVED** [review]

**Review by dalg24 — APPROVED** [review]

**Comment by tpadioleau** [comment]
@dalg24 Actually I think I got confused by other entries in the changelog 4.4, I don't think this one has ever been in it.

**Review by crtrott — CHANGES_REQUESTED** [review]
Why under bug fixes? Isn't this more a General Enhancement?

**Comment by tpadioleau** [comment]
> Why under bug fixes? Isn't this more a General Enhancement?

I added it in this section because we introduced the macro to fix existing user-defined deduction guides of `TeamThreadMDRange` in device code.

I don't know if we claimed to support in device code though :D

**Pull request merged**